client/packages/android/app/src/main/java/org/openmsupply/client/NativeApi.java

```diff
@@ -410,15 +413,39 @@ public void readLog(PluginCall call) {
         call.resolve(response);
     }
 
+    @PluginMethod()
+    public void acquireWakeLock(PluginCall call){
+        try {
+            PowerManager powerManager = (PowerManager) getContext().getSystemService(POWER_SERVICE);
+            wakeLock = powerManager.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK,"omSupply::WakelockTag");
+            wakeLock.acquire();
+        } catch (Exception e) {
+            Log.e(OM_SUPPLY, "Error acquiring wakelock: " + e.getMessage() );
```

Comment:
Maybe I am too paranoid but technically you should bubble the exception up other wise you could get out of sync with acquire and release calls, e.g. if acquire fails but release succeeds or the other way around...

Reply:
I didn't want the wake lock acquisition to cause an error in the application though - if the acquire fails, and the device then sleeps, that isn't an issue. We may then try to release a lock that doesn't exist, that isn't operationally a problem though. 

The problem would be if we are unable to release a lock that does exist! then we drain the battery quicker than we should. I still don't think the app should crash if that's the case..